mzLib/Readers/ExternalResults/SupportClasses/ModificaitonConverter.cs

```diff
@@ -0,0 +1,130 @@
+﻿using Omics.Modifications;
+using System.Collections.Concurrent;
+using System.Reflection;
+using UsefulProteomicsDatabases;
+
+namespace Readers;
+
+public static class ModificationConverter
+{
+    #region All Known Mods for MetaMorpheus - Use this to check for known mods to compare against 
+
+    // Cache of previously successful modification conversions to reduce the number of times the same modification is converted
+    private static readonly ConcurrentDictionary<(string, char), Modification> _modificationCache;
+
+    internal static List<Modification> AllKnownMods;
+
+    static ModificationConverter()
+    {
+        _modificationCache = new ConcurrentDictionary<(string, char), Modification>();
+
+        var info = Assembly.GetExecutingAssembly().GetName();
+        var name = info.Name;
+
+        var unimodStream = Assembly.GetExecutingAssembly().GetManifestResourceStream($"{name}.Resources.unimod.xml");
+        var unimodMods = UnimodLoader.ReadMods(unimodStream);
+
+        var psiModStream = Assembly.GetExecutingAssembly().GetManifestResourceStream($"{name}.Resources.PSI-MOD.obo.xml");
+        var psiModObo = Loaders.LoadPsiMod(psiModStream);
+        var formalChargeDict = Loaders.GetFormalChargesDictionary(psiModObo);
+
+        var uniprotStream = Assembly.GetExecutingAssembly().GetManifestResourceStream($"{name}.Resources.ptmlist.txt");
+        var uniprotMods = PtmListLoader.ReadModsFromFile(new StreamReader(uniprotStream), formalChargeDict, out _);
+
+        var modsTextStream = Assembly.GetExecutingAssembly().GetManifestResourceStream($"{name}.Resources.Mods.txt");
+        var modsTextMods = PtmListLoader.ReadModsFromFile(new StreamReader(modsTextStream), formalChargeDict, out _);
+
+        AllKnownMods = unimodMods.Concat(uniprotMods).Concat(modsTextMods).ToList();
```

Comment:
What about the PsiMods?

Reply:
psi mods appear to only be used to generate the formal charge dictionary in MetaMorpheus